jbpm-flow/src/main/java/org/jbpm/process/core/timer/impl/GlobalTimerService.java

```diff
@@ -80,7 +81,9 @@ private TimerJobFactoryManager initJobFactoryManager() {
 
     @Override
     public JobHandle scheduleJob(Job job, JobContext ctx, Trigger trigger) {
-        if (ctx instanceof ProcessJobContext) {
+        // start process job context it is not job per session is triggers even before 
+        // the process
+        if (ctx instanceof ProcessJobContext && !(ctx instanceof StartProcessJobContext)) {
```

Comment:
This kind of makes sense to me that the job is fired before the process is started so I think there is no reason to hold this in memory probably? If it is a timer in the middle of the process instance, then it can be useful to "track" progress, but otherwise...

So what is really the point of saving this in memory is not clear to me right now. Maybe it is serialized somewhere?

Reply:
pls read my comment explaining this.

Reply:
I understand what you said in the comment. My questions was: "do we need to keep it in memory for normal ProcessJobContext as well? Because by looking at the code, it is only needed for "check duplicated purposes as far as I can tell. See: https://github.com/kiegroup/jbpm/blob/186692eb6c88942a82a754147d4c2eab54939ab9/jbpm-flow/src/main/java/org/jbpm/process/core/timer/impl/GlobalTimerService.java#L86-L98

Just wanted to know what is its purpose. That's all.